### ISSUE TITLE
issue with useEffect running on redirect even with a conditional that should stop it

### ISSUE THREAD

**Issue by nickesparza** [sei-ec-remote/team-project-issues#68]
**Describe the bug**
Our app requires all of our routes, including the GET routes, to be authenticated. However, when logging out, the Home page still tries to run useEffect, which then throws an error message on the front page instead of showing the "logged out" state.

**What is the problem you are trying to solve?**
We want the home route to show a splash page when the user is not logged in, and the list of documents when they are. When they log out, we want it to go back to the splash page instead of getting stuck in the "loading" state like it is now.

**Expected behavior**
User accesses the site and sees a default page with no information. They log in and the page shows all our documents. They log out, and it returns to the default page with no information

**What is the actual behavior?**
Upon logout, the Loading spinner appears and a msgAlert pops that says "Error fetching patients." This is expected because the fetch route needs a user token and there is none after they logged out.

**Post any code you think might be relevant (one fenced block per file)**
This is how we're checking and running useEffect right now. It works on first load but does not if they get back to this page via sign-out.
```js
const PatientIndex = (props) => {
    const [patients, setPatients] = useState(null)
    const { msgAlert } = props
    const { user } = props
    const navigate = useNavigate()
    console.log('user in PatientIndex', user)
    useEffect(() => {
        console.log('useEffect has run')
        if (user) {
            getAllPatients(user)
            // .then(res => console.log(res))
            .then(res => setPatients(res.data.patients))
            //console.log('use effects work')
            .catch(err => {
                msgAlert ({
                heading: 'Error Getting Patients',
                message: messages.getPatientsFailure,
                variant: 'danger',
            })})
        // if there is no user, console log this message.. no showing the patients....
        } else {
            setPatients([])
           console.log('not logged in')
        }
    }, [])

    // this is fine as the loading screen condition
    if(!patients) {
        // return <p>Loading...</p>
        return <LoadingScreen />
    // if the patient list is 0, but a user exists, show them that they have the ability to add patients
    } else if (patients.length === 0 && user ) {
        return <p>No patients yet...Lets Add some.</p>
    // if there is no user, tell them to log in
    } else if (!user) {
        return <p>please log in to see patients.</p>
    }
```

**What is your best guess as to the source of the problem?**
I'm not really sure but my guess is that because useEffect is set to run no matter what on component load, it tries to run but can't because user is null. That doesn't explain why the if statement isn't catching this though.

**What things have you already tried to solve the problem?**
I've tried fiddling with the conditionals on what shows up in what state of user, and briefly entertained adding a "loggedIn" state to the main page, but that feels like overkill. I'm thinking we may want to break the document index and the splash page into two separate components and have sign-out redirect straight to the splash rather than trying to reload Home.

**Additional context**

**Paste a link to your repository here**
https://github.com/nickesparza/ga-project-3-client

**Comment by timmshinbone** [comment]
Have you tried adding the `user` to the dependency array? Not sure why the if isn't working out, but try that and let me know if the behavior changes.

**Comment by nickesparza** [comment]
I did try that, but as far as I can tell it's failing in the same way (a 401 Unauthorized error). This is the weirdest part because the console definitely shows a null user but the if (user) statement in useEffect is being ignored.
![image](https://user-images.githubusercontent.com/8571987/183136689-af4114dd-a528-4493-baea-8dec86714957.png)

**Comment by nickesparza** [comment]
I figured out what was happening.

Because of the order of how SignOut was handling things, it was navigating to the home page before setting the user to null - hence, useEffect would start to run, see that user was null, and throw a 401 error. I changed the order of things and then ran into an issue where it didn't redirect to the home page, but rather to the sign in. the RequireAuth component was set to redirect ANY page to /sign-in when it detected a null user, so I changed that redirect to '/' and that has fixed it for now.
